portal-impl/src/com/liferay/portal/db/partition/DBPartitionUtil.java

```diff
@@ -150,7 +150,9 @@ public static boolean isPartitionEnabled() {
 	public static void populateCompanyId(
 		Class<?> entityClass, BaseModel<?> baseModel) {
 
-		if (!_DATABASE_PARTITION_ENABLED) {
+		if (!GetterUtil.getBoolean(
```

Comment:
Hey @shuyangzhou, as I explained in the commit, I changed this because for the DBPartitionUtilTest we startup the server with db_partition_enabled to false and we change it on the fly. However, changing the constant is not enough, it seems the compiler already cached the value for these static methods, I also don't like this solution but do you have a suggestion to fix it?

Reply:
This is no real solution for it with the current design of database partition. So either the design must be refreshed with testing in mind, or QA has to create a dedicated batch for it where it is enabled.